digester/main.go

```diff
@@ -19,6 +66,65 @@ func CheckError(err error) {
 	}
 }
 
+func unzipFile(file *zip.File, dest string) error {
+	// Check if file paths are not vulnerable to Zip Slip
+	filePath := filepath.Join(dest, file.Name)
+	if !strings.HasPrefix(filePath, filepath.Clean(dest)+string(os.PathSeparator)) {
+		return fmt.Errorf("%s: illegal file path", filePath)
+	}
+
+	if file.FileInfo().IsDir() {
+		if err := os.MkdirAll(filePath, os.ModePerm); err != nil {
```

Comment:
Why are we trying to create this directory?

Reply:
it will create any directory required for that filepath to be valid.

For e.g. if the first file is suppose `/slack-export/general/2015-05-05.json` then it will create the `slack-export` directory and inside it the `general` directory.

Reply:
Ah ok